src/activations.jl

```diff
@@ -256,32 +255,17 @@ about twice the error of Julia's, about twice the speed
     ifelse(x2 < 66f0, x * (n / d), sign(x))
 end
 
-# tanh_fast(x::Float16) # probably better. tanh_fast(x::Real) is very badly behaved for Float16.
-
-@inline function tanh_fast(x::Float16)
-    x2 = abs2(x)
-    n = evalpoly(x2, (Float16(1.0), Float16(0.1346), Float16(0.003597), Float16(2.235e-5)))
-    d = evalpoly(x2, (Float16(1.0), Float16(0.468), Float16(0.02626), Float16(0.0003455), Float16(9.0e-7)))
-    ifelse(x2 < Float16(18), x * (n / d), sign(x))
-end
-
-"""
-    tanh_fast(x) = exp(2x) / (1 + exp(2x))
-
-This is faster, but slightly less accurate, version of `tanh`.
-For `x::Float64`, this has errors up to about `3e-15`, sometimes `> 10 * eps(x)`,
-and should be about 2-5 times faster.
-
-But `tanh_new(x::Float32)` is more important, and a different approximation.
-"""
 @inline function tanh_fast(x::Real)
-   exp2x = @fastmath exp(x + x)
-   y = (exp2x - 1) / (exp2x + 1)
-   ifelse(x > 30, one(y), ifelse(x < -30, -one(y), y))
+    exp2x = @fastmath exp(x + x)
+    y = (exp2x - 1) / (exp2x + 1) 
+    # That has bad errors near zero; using expm1 would be slower & more accurate.
+    # Instead, we switch to an taylor series; seems to add about 50% to time.
+    x2 = x * x
+    ypoly = x * evalpoly(x2, (0.9999999999999999, -0.33333333333309806, 0.13333333318143492, -0.053968217983868146 , 0.021865628148606587, -0.008671836868790176))
```

Comment:
How do these coeffs do? `(1.0, 0.33333333333333337, 0.1333333333332623, -0.0539682539194502, 0.021869476267930975, -0.00886184042142138, 0.0035188503873932893)` In general, I've found that forcing the leading coefficients to exact values can help with error since you remove the error in the coeficient. These were generated with the following code.
```
coefs = [1/big(3) , 2/big(15) ,- 17/big(315) , 62/big(2835) ,- 1382/big(155925) , 21844/big(6081075) ,- 929569/big(638512875) , 6404582/big(10854718875) ,- 443861162/big(1856156927625)]

julia> g(x)=evalpoly(x, coefs)

julia> Tuple(Float64.(ratfn_minimax(g, (1e-15,.017^2),6,0)[1]))
```

Reply:
That is better (with -0.3333 surely), indeed. In fact this looks to be within 1eps within the range for which I used the polynomial:
```
julia> maximum(x -> abs(tanh(big(x)) - tanh_fast(x)) / eps(Float64(tanh(big(x)))), 0.000_01:0.000_01:0.13) |> Float64  0.9878056794145185  # with yours
1.8541809732394265  # with mine
```
Whereas the point which is 5eps is somewhere `0.2<x<0.3`. So perhaps the ideal changeover point is quite a bit larger... I think 0.13 was a very quick guess at where the error got to 4eps or so. I'm not being maximally systematic about this, clearly.

Reply:
The other option is to lower the degree polynomial. The one I gave is a minimax polynomial, so it won't be anywhere near optimal outside the range. Do you have an error threshold in mind? If so, I'll do some searching to find the minimal degree one with the required accuracy of the range.

Reply:
I vaguely had 5eps in mind, just because that was doable for the Float32 rational function --- there I could fit more more terms, but after rounding etc. couldn't do much better. I'm not sure how much NNlib cares about the Float64 case, it might just be for our amusement. But sure, if you are set up to search then a better polynomial + threshold would be welcome. 

BTW I tried replacing `exp(x+x)` with `exp2(α x)` as suggested, but didn't see much change in speed. The accuracy does seem a little worse on one side, so perhaps it wants to be `exp(abs(2x))` & restore the sign later.

Reply:
So, it looks like to maintain 5 ULPs, we need a polynomial for `|x| < 0.063`. Given that, we can lower the degree to 4, which gives coefficients of `(1.0,  -0.3333333333331805,  0.13333333295910244, -0.05396796907132604,  0.02178397571543739)` which have a maximal error of 1.2 ULPs in the range. Alternatively, bumping up to degree 5 lets the polynomial stay accurate to 2.5 ULPs over the whole range with coefficients `(1.0, -0.33333333333324583, 0.13333333325511604, -0.05396823125794372, 0.02186660872609521, -0.008697141630499953)`. I would probably go for this version, since the 1 extra fma won't change performance too much, and will give 2x less accuracy, which imo is worth it.